vault/raft.go

```diff
@@ -940,6 +940,32 @@ func (c *Core) JoinRaftCluster(ctx context.Context, leaderInfos []*raft.LeaderJo
 				}
 
 				for _, addr := range addrs {
+					u, err := url.Parse(addr)
+					if err != nil {
+						c.logger.Info("failed to parse discovered address", "error", err)
+						continue
+					}
+
+					if u.Scheme == "" {
+						scheme := leaderInfo.Scheme
+						if scheme == "" {
+							// default to HTTPS when no scheme is provided
+							scheme = "https"
+						}
+
+						addr = fmt.Sprintf("%s://%s", scheme, addr)
```

Comment:
Why not just populate `u`'s members, then call `String()` on it?

Reply:
We can also do that :)